File: src/main/java/org/nryotaro/httpcli/handler/CliHandler.java
package org.nryotaro.httpcli.handler;


import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.http.HttpContent;
import io.netty.handler.codec.http.HttpResponse;
import io.netty.handler.codec.http.LastHttpContent;

public interface CliHandler {

    void onFailure(Throwable cause);

    void acceptHttpResponse(HttpResponse response);

    void acceptContent(HttpContent msg);

    void acceptLastHttpContent(LastHttpContent msg);

    void onException(ChannelHandlerContext ctx , Throwable cause);
}

File: src/main/kotlin/org/nryotaro/handler/CliHandler.kt
package org.nryotaro.handler

import io.netty.channel.ChannelHandlerContext
import io.netty.handler.codec.http.HttpContent
import io.netty.handler.codec.http.HttpResponse
import io.netty.handler.codec.http.LastHttpContent

interface CliHandler {

    fun onFailure(cause: Throwable)

    fun acceptHttpResponse(response: HttpResponse)

    fun acceptContent(msg: HttpContent)

    fun  acceptLastHttpContent(msg: LastHttpContent)

    fun  onException(ctx: ChannelHandlerContext, cause: Throwable)

}
